src/smsCommandFields/FieldDataElementWithCategoryOptionComboAddFormulaButton.js

```diff
@@ -0,0 +1,82 @@
+import { Button, ReactFinalForm } from '@dhis2/ui'
+import { PropTypes } from '@dhis2/prop-types'
+import { useDataEngine } from '@dhis2/app-runtime'
+import React, { useEffect, useState } from 'react'
+import i18n from '../locales'
+import styles from './FieldDataElementWithCategoryOptionComboAddFormulaButton.module.css'
+
+const { useField } = ReactFinalForm
+
+const DATA_ELEMENTS_QUERY = {
+    dataElement: {
+        resource: 'dataElements',
+        params: ({ code }) => ({
+            filter: `code:eq:${code}`,
+            fields: 'displayName',
+            paging: 'false',
+        }),
+    },
+}
+
+export const FieldDataElementWithCategoryOptionComboAddFormulaButton = ({
```

Comment:
We're replicating a lot of what the dataengine does internally here. I wonder if we could handle this differently eventually to let the dataengine do the heavy lifting.